bundles/http/Citrix/Covid-19-Self-Certify/script.js

```diff
@@ -0,0 +1,219 @@
+            dataStore.save("items", {
+                "employeeName": getTextValue(item.fields, 'text', 'Name') ?? null,
+                "email": getTextValue(item.fields, 'email') ?? null,
+                "confirmation": getCategoryValue(item.fields) ?? null,
+                "createdOn": getTimestamp(item.fields) ?? null,
+                "app_item_id": item.app_item_id,
+                "region": getTextValue(item.fields, 'text', 'Region') ?? null,
+                "country": getTextValue(item.fields, 'text', 'Country') ?? null,
+            })
+        })
+        offset += limit
+    } while (dataUpdateAfterAction != true && offset <= total)
+}
+
+async function covidSelfCertify({ client, dataStore, integrationParameters, actionParameters }) {
+    if (actionParameters.appId !== null) {
+        dataStore.deleteById('items', actionParameters.email)
+    }
+    const response = await client.fetch(`item/app/${integrationParameters.covidAppId}`, {
+        method: "POST",
+        body: JSON.stringify({
+            "fields": {
+                "email": [{ "type": "work", "value": actionParameters.email }],
+                "title": [{ "value": actionParameters.email.substring(0, actionParameters.email.indexOf('@')).split('.').reduce((initialvalue, currentvalue) => { return initialvalue + (currentvalue[0].toUpperCase() + currentvalue.substring(1, currentvalue.length) + ' ') }, '').trimEnd() }],
+                "confirmation": actionParameters.status,
+                "createdon": {
+                    "start": moment().format('YYYY-MM-DD HH:mm:ss')
+                },
+                "country": [{ "value": actionParameters.country }],
+                "city": [{ "value": actionParameters.city }]
+            }
+        })
+    })
+    if (!response.ok) {
+        throw new Error(await response.text())
+    }
+    await getItems(client, dataStore, integrationParameters, true)
+}
+
+async function getRegionData(client, dataStore, integrationParameters) {
+    const limit = 100
+    let offset = 0
+    let total = 0
+    do {
+        const countryDataResponse = await client.fetch(`item/app/${integrationParameters.countryDataAppId}/filter`, {
+            method: "POST",
+            body: JSON.stringify({
+                limit: limit,
+                offset: offset
+            })
+        })
+        const {responseBody,errorMessage,errorStatus} = await validateResponse(countryDataResponse)
+        if (errorStatus) {
+            console.log(JSON.stringify(responseBody),errorMessage)
+            throw new Error(errorMessage)
+        }
+        total = responseBody.total
+        const items = responseBody.items ?? []
+        items.forEach(item => {
+            const fields = item.fields ?? []
+            const region = getTextValue(item.fields, 'text', 'Region') ?? null
+            const countries = String(getTextValue(item.fields, 'text', 'Countries')).split(',')
+            countries?.forEach(country => {
```

Comment:
if some `item` has the same countries will it create duplicate countries records or not?

Reply:
We have used `uniq` function to remove the duplicate countries.

Reply:
it's unique per `item.fields` but not per all `items`, right?